packages/avatars/src/styled/StyledStatusIndicator.ts

```diff
@@ -0,0 +1,142 @@
+
+import { IAvatarProps, SIZE, STATUS } from '../types';
+import { getStatusColor } from './utility';
+
+export interface IStatusIndicatorProps extends Omit<IAvatarProps, 'badge' | 'isSystem' | 'status'> {
+  readonly status?: IAvatarProps['status'] | 'active';
+  borderColor?: string;
+}
+
+const COMPONENT_ID = 'avatars.status-indicator';
+
+const [xxs, xs, s, m, l] = SIZE;
+const [active, available, away, transfers, offline] = ['active', ...STATUS];
+
+const sizeStyles = (props: IStatusIndicatorProps & ThemeProps<DefaultTheme>) => {
+  let visible = true;
+  let height = '0';
+  let borderWidth = props.theme.shadowWidths.sm;
+
+  switch (props.size) {
+    case xxs:
+      visible = false;
+      borderWidth = math(`${borderWidth} - 1`);
+      height = math(`${props.theme.space.base}px - ${borderWidth}`);
+      break;
+    case xs:
+      visible = false;
+      height = math(`${props.theme.space.base * 2}px - (${borderWidth} * 2)`);
+      break;
+    case s:
+    case m:
+    case l:
+      if (props.size === s && props.status !== 'active') {
+        height = math(`${props.theme.space.base * 3}px - (${borderWidth} * 2)`);
+      } else {
+        height = math(`${props.theme.space.base * 4}px - (${borderWidth} * 2)`);
+      }
+      break;
+  }
+
+  return css`
+    border: ${borderWidth} ${props.theme.borderStyles.solid};
+    border-radius: ${height};
+    min-width: ${height};
+    max-width: 2.48em;
+    height: ${height};
+    box-sizing: content-box;
+    overflow: hidden;
+    text-align: center;
+    text-overflow: ellipsis;
+    line-height: ${height};
+    white-space: nowrap;
+    font-size: ${props.theme.fontSizes.xs};
+    font-weight: ${props.theme.fontWeights.semibold};
+
+    & > span {
+      padding: 0 ${math(`${props.theme.space.base + 1}px - (${borderWidth})`)};
+    }
+
+    & > svg {
+      width: 100%;
+      height: 100%;
+
+      & circle {
+        display: none;
+      }
+    }
+
+    ${!visible &&
+    css`
+      & > * {
+        display: none;
+      }
+    `}
+  `;
+};
+
+const colorStyles = (props: IStatusIndicatorProps & ThemeProps<DefaultTheme>) => {
+  const foregroundColor = props.foregroundColor || props.theme.palette.white;
+  const surfaceColor =
+    props.surfaceColor ||
+    (props.status ? props.theme.colors.background : (props.theme.palette.white as string));
+  let backgroundColor = props.backgroundColor || 'transparent';
+  let borderColor = props.borderColor || backgroundColor;
+  let boxShadow = props.theme.shadows.sm(surfaceColor);
+
+  if (xxs === props.size) {
+    boxShadow = boxShadow.replace(props.theme.shadowWidths.sm, '1px');
+  }
+
+  switch (props.status) {
+    case available:
+    case active:
+    case away:
+    case transfers:
+      backgroundColor = getStatusColor(props.status, props.theme);
+      borderColor = backgroundColor;
+      break;
+    case offline:
+      borderColor = getStatusColor(props.status, props.theme);
+      backgroundColor = surfaceColor;
```

Comment:
Using `surfaceColor` as the background color for offline feels wrong. My hunch is that the design wants this to be `props.theme.palette.white` in every case, no matter what surface it is sitting on.